src/test/scala/cognite/spark/v1/FilesRelationTest.scala

```diff
@@ -49,64 +50,82 @@ class FilesRelationTest extends FlatSpec with Matchers with SparkTest {
       .load()
     df.createOrReplaceTempView("filesSource")
 
+    //insert data
     spark
       .sql(s"""
-            |select id,
-            |name,
-            |source,
-            |id as externalId,
-            |mimetype,
-            |metadata,
-            |Array(1739842716040355) as assetIds,
-            |uploaded,
-            |uploadedTime,
-            |createdTime,
-            |lastUpdatedTime,
-            |uploadUrl,
-            |dataSetId
-            |from filesSource
-            |where source = '$originalSource'
+            |select "ex" as name,
+            |'$originalSource' as source,
+            |'upsert-$originalSource' as externalId,
+            |null as id
      """.stripMargin)
-      .select(df.columns.map(col): _*)
       .write
-      .insertInto("filesSource")
+      .format("cognite.spark.v1")
+      .option("apiKey", writeApiKey)
+      .option("type", "files")
+      .option("onconflict", "upsert")
+      .save()
+    val id = writeClient.files.retrieveByExternalId(s"upsert-$originalSource").id
 
-    val dfWithTestDirectory = retryWhile[List[File]](
-      writeClient.files.filter(FilesFilter(assetIds = Some(Seq(1739842716040355L)))).compile.toList,
-      files => files.length < 10)
-    assert(dfWithTestDirectory.length == 10)
+    val insertWithUpsertIds =
+      retryWhile[Array[Row]](
+        spark
+          .sql(s"select id from filesSource where source = '$originalSource'")
+          .collect,
+        df => df.length < 1)
+    assert(insertWithUpsertIds.length == 1)
 
+    //update data
     spark
       .sql(s"""
-            |select id,
-            |name,
-            |source,
-            |id as externalId,
-            |mimetype,
-            |metadata,
-            |Array(176631399357975) as assetIds,
-            |uploaded,
-            |uploadedTime,
-            |createdTime,
-            |lastUpdatedTime,
-            |uploadUrl,
-            |dataSetId
-            |from filesSource
-            |where source = '$originalSource'
+            |select ${id.toString} as id,
+            |'updated-$originalSource' as source,
+            |"ex" as name,
+            |'updated-externalId-$originalSource' as externalId
      """.stripMargin)
-      .select(df.columns.map(col): _*)
       .write
-      .insertInto("filesSource")
+      .format("cognite.spark.v1")
+      .option("apiKey", writeApiKey)
+      .option("type", "files")
+      .option("onconflict", "upsert")
+      .save()
```

Comment:
We should still have at least some tests that still use `.insertInto`, as that's a different interface and uses different code. Better to just add a separate test for `"onconflict", "upsert"`, I think.